src/hatti/views/dataview.cljs

```diff
@@ -87,6 +87,42 @@
                              {:selected view})
       (put! shared/event-chan {:re-render view}))))
 
+(defn activate-settings-view! [view settings-section]
+  (let [view (keyword view)
+        settings-section (keyword settings-section)
+        views (-> @shared/app-state :views :all)
+        settings-views (-> @shared/app-state :views :settings :all)]
+    (when (contains? (set settings-views) settings-section)
+      (merge-into-app-state! shared/app-state
+                             [:views]
+                             {:selected view})
+      (merge-into-app-state! shared/app-state
+                             [:views :settings]
+                             {:active-tab settings-section})
+      (put! shared/event-chan {:re-render view}))))
+
+(defn activate-integrated-apps-view! [view settings-section app-type]
+  (let [view (keyword view)
+        active-section (keyword app-type)
+        settings-section (keyword settings-section)
+        views (-> @shared/app-state :views :all)
+        settings-views (-> @shared/app-state :views :settings :all)]
+    (when (contains? (set settings-views) settings-section)
+      (merge-into-app-state! shared/app-state
+                             [:views]
+                             {:selected view})
+      (merge-into-app-state! shared/app-state
+                             [:views :settings]
+                             {:active-tab settings-section})
+      (merge-into-app-state! shared/app-state
+                             [:views :settings :integrated-apps]
+                             {:active-section active-section})
+      (if (= app-type "google_sheets")
```

Comment:
Mmmh I don't think we can avoid the special case. The reason for this is that setting `{add? :true}` will show the google-form which is what we want when we redirect the user after google auth but for the other apps we don't necessarily need to show the form. But definitely we can use something better than a string for the condition.